fix(main menu): contrast, icons, responsiveness of main menu

mndonx opened the pull request:
## Summary

<!-- Summary of the PR -->

This PR fixes/implements the following **bugs/features**
See: https://emulsify-ds.github.io/compound/?path=/story/molecules-menus--main

- Main menu contrast in light & dark mode does not past testing
<img width="966" alt="Screen Shot 2023-01-17 at 4 21 25 PM" src="https://user-images.githubusercontent.com/2343957/213025497-253a2413-4284-4e7b-9c6c-7729a1efbc13.png">

- Mobile menu icon is broken 
<img width="571" alt="Screen Shot 2023-01-17 at 4 21 48 PM" src="https://user-images.githubusercontent.com/2343957/213025523-0dd4ef8b-643e-4f7a-89e1-f1f77e2d039f.png">

- Close icon should match open icon (uses a character instead of an SVG)
- Main menu in mobile has weird padding and margins
<img width="568" alt="Screen Shot 2023-01-17 at 4 21 56 PM" src="https://user-images.githubusercontent.com/2343957/213025554-c1cca8cf-0e37-4a80-a524-71af0f11c05a.png">

- Mobile main menu should align right (to allow for logo)
- Main menu is not responsive (can't go from small with open items to large and have the menu work)
<img width="1021" alt="Screen Shot 2023-01-17 at 4 22 29 PM" src="https://user-images.githubusercontent.com/2343957/213025626-1d21bb01-48bf-4367-8d31-415f7db91438.png">

Related:
- Logo is not displaying at small widths
- Social icons not displaying
<img width="508" alt="Screen Shot 2023-01-17 at 4 23 06 PM" src="https://user-images.githubusercontent.com/2343957/213025660-db194d8a-8142-4632-befe-368ce667bcde.png">


<!-- You can skip this if you're fixing a typo or adding an app to the Showcase. -->

I starting fixing stuff because the bare minimum of contrast testing was not met ... but then I found some other stuff. However, the Main Menu is currently not accessible (the second level) so I think really more work will need to be put in here. I think this is just a stop-gap. 

<!-- Example: When "Adding a function to do X", explain why it is necessary to have a way to do X. -->

## Documentation update (required)

If this pull request requires a change to Emulsify documentation, those changes, updates, and/or new information must accompany this pull request.

## How to review this pull request
<!-- Provide step-by-step instructions to functionally test this PR. -->
<!-- Ensure that your code passing the repo's linting standards. -->
- [ ] After recompiling, check the Main Menu component (also check the Site Header and Article content type (and full Pages) 
- [ ] Make sure the contrast is good
- [ ] Make it small to check the mobile version. Make sure that looks good (especially on Content Type and Page versions)
- [ ] Make sure all images are loading properly in Main Menu and Social Menu (as well as the Logo at small widths.)
- [ ] Make sure when going from small (with the menu open) to large, the menu continues to work

## Closing issues

<!-- Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such). -->

Closes #81

ryanhagerty commented on components/02-molecules/menus/main-menu/_03-main-menu-toggle.scss:
```diff
@@ -2,19 +2,23 @@
+  width: 100vw;
   text-decoration: none;
-  width: 100%;
+  top: 1em;
```
Let's use `rem` throughout to make it consistent

ryanhagerty commented on components/02-molecules/menus/main-menu/_00-main-menu.scss:
```diff
@@ -15,9 +15,18 @@ $main-menu-medium: $medium;
+      position: relative;
+      top: auto;
+      right: auto;
+      width: auto;
```
Let's adjust the ordering of items here to match the rest of the file / be in sync with how we're doing it going forward.

ryanhagerty commented on components/02-molecules/menus/main-menu/_00-main-menu.scss:
```diff
@@ -47,7 +56,7 @@ $main-menu-medium: $medium;
 
   @include medium {
-    background-color: clr(highlight-high);
+    background-color: clr(accent-high);
```
Same here

ryanhagerty reviewed (changes requested):
Great work, Amanda. Just a few minor changes and this should be GTG. 🏅 

Looking more closely, there are some strange choices that were preexisting. I think it's a side effect of constructing a skeleton for Compound rather than something fully fleshed out, but there's a couple areas we could take this further without dipping our toes into making this opinionated.

mndonx commented:
@ryanhagerty Thank you for the review! I've adjusted the order of the CSS throughout the component and changed some ems to rems. I also swapped the `:focus` to `:focus-within` so people tabbing through the main menu in desktop can actually get to the second level. Not a perfect a11y solution, but better than what it is now!

ryanhagerty reviewed (approved):
LGTM - good job 🚂 

There's a couple things I noticed that I think can be in a follow up task that were pre-existing. So feel free to ignore the below, but wanted to call it out for compound as a whole:

1. there's lots of mixing of `rem` and `px`. We should go ahead make a group decision on when to apply what and where
2. Variables like `$space`. They get used sometimes, but other times I see `1rem` or even `16px`. Additionally, `$space` is used in different contexts. In my mind, `$space` should be reserved for vertical rhythm, like between components, headings, etc. **not** for borders, padding, gaps, etc. This is an example of reusing for the sake of reusing in my mind.
3. Some components like this menu rely on assumptions about the height of the component like `top: 110px`. Anyone using this component will have to refactor that, so we should make these kind of things magic number free.

Thank you for attending my unrelated rant.

mndonx commented:
@ryanhagerty I totally agree, except I'm fairly sure this component is going to be completely rewritten for the UI Kit (using a very different method) and back-ported here, so I probably already spent too much time on it already -- I think it's just making sure it isn't completely broken right now. Maybe we should create an issue for fixing inconsistencies in rem/$space/px throughout the project, though? I'll do that, and set this one ready to merge.

mcortes19 commented:
@mndonx Could you please review/fix conflicts, thanks for all you effort 💪🏼

mndonx commented:
@mcortes19 I fixed the conflicts (sorry about the delay!) so I think it is ready for merge. I'm not sure why the checks do not complete -- but it seems like that is the case on other PRs?
